src/main/scala/com.gu.contentapi.client/utils/CapiModelEnrichment.scala

```diff
@@ -15,4 +14,46 @@ object CapiModelEnrichment {
     def toCapiDateTime: CapiDateTime = CapiDateTime.apply(dt.toInstant.toEpochMilli, DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(dt))
   }
 
+  implicit class RichContent(val content: Content) extends AnyVal {
+
+    def designType: DesignType = {
+
+      val defaultDesignType = Article
+
+      def isImmersive(c: Content) = c.fields.flatMap(_.displayHint).map(_.equals("immersive")).getOrElse(false)
+
+      def tagExistsWithId(tagId: String)(c: Content) = c.tags.exists(tag => tag.id == tagId)
+
+      def isMedia(c: Content) = tagExistsWithId("type/audio")(c) || tagExistsWithId("type/video")(c) || tagExistsWithId("type/gallery")(c)
+
+      def isReview(c: Content) = tagExistsWithId("tone/reviews")(c) || tagExistsWithId("tone/livereview")(c) || tagExistsWithId("tone/albumreview")(c)
+
+      def isComment(c: Content) = tagExistsWithId("tone/comment")(c) || tagExistsWithId("tone/letters")(c)
+
+      type DesignTypeFilter = Content => Option[DesignType]
+
+      def filter(result: Option[DesignType])(contentFilter: Content => Boolean) : DesignTypeFilter = {
+        c => if(contentFilter(c)) result else None
+      }
+
+      val filters: List[DesignTypeFilter] = List(
+        filter( Some(Immersive) ) ( isImmersive ),
+        filter( Some(Media) ) ( isMedia ),
+        filter( Some(Review) ) ( isReview ),
+        filter( Some(Analysis) ) ( tagExistsWithId("tone/analysis") ),
+        filter( Some(Comment) ) ( isComment ),
+        filter( Some(Feature) ) ( tagExistsWithId("tone/features") ),
+        filter( Some(Live) ) ( tagExistsWithId("tone/minutebyminute") )
+      )
+
+      def findDesignType(contentFilters: List[DesignTypeFilter], c: Content) : DesignType = contentFilters match {
```

Comment:
If you change `isMedia` etc to vals then you could do something a tiny bit shorter, e.g.:
```
val isMedia: Content => Boolean = c => tagExistsWithId("type/audio")(c) || tagExistsWithId("type/video")(c) || tagExistsWithId("type/gallery")(c)
...
val predicates: Map[Content => Boolean, DesignType] = Map(
  isImmersive -> Immersive,
  isMedia -> Media,
  isReview -> Review,
  tagExistsWithId("tone/analysis") -> Analysis,
  isComment -> Comment,
  tagExistsWithId("tone/features") -> Feature,
  tagExistsWithId("tone/minutebyminute") -> Live
)

val result = predicates.collectFirst { case (predicate, design) if predicate(content) => design }
result.getOrElse(defaultDesignType)
```

Reply:
That's cool .. :)

Reply:
Except i can't get it to compile . Might give you a shout tomorrow mornign

Reply:
Oops, didn't realise the order is important. The following works locally for me, what do you think? Happy to chat tomorrow!
```
def designType: DesignType = {

  val defaultDesignType = Article

  val isImmersive: Content => Boolean = c => c.fields.flatMap(_.displayHint).exists(_.equals("immersive"))
  def tagExistsWithId(tagId: String): Content => Boolean = c => c.tags.exists(tag => tag.id == tagId)
  val isMedia: Content => Boolean = c => tagExistsWithId("type/audio")(c) || tagExistsWithId("type/video")(c) || tagExistsWithId("type/gallery")(c)
  val isReview: Content => Boolean = c => tagExistsWithId("tone/reviews")(c) || tagExistsWithId("tone/livereview")(c) || tagExistsWithId("tone/albumreview")(c)
  val isComment: Content => Boolean = c => tagExistsWithId("tone/comment")(c) || tagExistsWithId("tone/letters")(c)

  val predicates: List[(Content => Boolean, DesignType)] = List(
    isImmersive -> Immersive,
    isMedia -> Media,
    isReview -> Review,
    tagExistsWithId("tone/analysis") -> Analysis,
    isComment -> Comment,
    tagExistsWithId("tone/features") -> Feature,
    tagExistsWithId("tone/minutebyminute") -> Live
  )

  val result = predicates.collectFirst { case (predicate, design) if predicate(content) => design }
  result.getOrElse(defaultDesignType)
}
```